src/main/java/de/adesso/anki/battle/world/bodies/Rocket.java

```diff
@@ -1,24 +1,45 @@
 package de.adesso.anki.battle.world.bodies;
 
+import java.util.Iterator;
 import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
 
 import org.eclipse.paho.client.mqttv3.MqttException;
 
 import com.states.GameState;
 
 import de.adesso.anki.battle.mqtt.MqttService;
+import de.adesso.anki.battle.util.Position;
+import de.adesso.anki.battle.world.Body;
 import de.adesso.anki.battle.world.DynamicBody;
+import de.adesso.anki.battle.world.World;
 
 public class Rocket extends DynamicBody {
-
+//TODO should rockets follow the roadmap ? 
+	// else explode after x seconds for garbage collector
 	private String direction; 
 	
+	// hotfix for driving into own rockets
+	private long timer ; 
+	private boolean active;
 	
 	public Rocket (String direction) {
 		this.direction = direction;
+		timer = System.currentTimeMillis();
 	}
 	
 	
+	
+	// TODO adjust timers

```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)